lib/spack/llnl/util/filesystem.py

```diff
@@ -440,35 +444,154 @@ def fix_darwin_install_name(path):
                         stdout=subprocess.PIPE).communicate()[0]
                     break
 
+# Utilities for libraries
 
-def to_lib_name(library):
-    """Transforms a path to the library /path/to/lib<name>.xyz into <name>
+
+class LibraryList(collections.Sequence):
+    """Sequence of absolute paths to libraries
+
+    Provides a few convenience methods to manipulate library paths and get
+    commonly used compiler flags or names
     """
-    # Assume libXYZ.suffix
-    return os.path.basename(library)[3:].split(".")[0]
 
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
 
-def to_link_flags(library):
-    """Transforms a path to a <library> into linking flags -L<dir> -l<name>.
+    @property
+    def directories(self):
+        """Stable de-duplication of the directories where the libraries
+        reside
 
-    Return:
-      A string of linking flags.
-    """
-    dir  = os.path.dirname(library)
-    name = to_lib_name(library)
-    res = '-L%s -l%s' % (dir, name)
-    return res
+        >>> l = LibraryList(['/dir1/liba.a', '/dir2/libb.a', '/dir1/libc.a'])
+        >>> assert l.directories == ['/dir1', '/dir2']
+        """
+        return list(dedupe(
+            os.path.dirname(x) for x in self.libraries if os.path.dirname(x)
+        ))
+
+    @property
+    def libnames(self):
```

Comment:
not sure if you missed my comment, but i think `libnames` name could be confusing for package authors in that they may expect it behave like `names` (i.e. return `FOO` from `libFOO.so`).

I think confusion could be avoided if you call it `basenames`.